rest/rest-resources/src/main/java/org/apache/brooklyn/rest/resources/AbstractBrooklynRestResource.java

```diff
@@ -49,12 +50,18 @@
 
     
     private ManagementContext managementContext;
+    private ManagementContextInternal managementContextInternal;
```

Comment:
`ManagementContextInternal` implements `ManagementContext` so it can be used at any place where `ManagementContext` is needed. And the code assumes that `ManagementContext` and `ManagementContextInternal` are backed by the same instance. So still think that having a single service and a single point to inject it is the way to go.